video/out/cocoa_cb_common.swift

```diff
@@ -122,7 +122,41 @@ class CocoaCB: Common {
 
         libmpv.setRenderICCProfile(colorSpace)
         if #available(macOS 10.11, *) {
-            layer?.colorspace = colorSpace.cgColorSpace
+
```

Comment:
i would most likely move the whole color space functinality added here into its own function(s) in video/out/mac/common.swift. for several reasons:
- it uses a "mac" prefixed option that should be available for all common mac code, not just the gl_layer and cocoa_cb
- makes it reusable
- easier to maintain

probably best to implement part of this functionality here https://github.com/mpv-player/mpv/blob/master/video/out/mac/common.swift#L397-L399 and call the super.updateICCProfile() in the cocoa-cb specific part.

Reply:
But isn't this functionality `GLLayer`-specific (which is only seems to be available in cocoa_cb)?

Reply:
the colorspace parameter is the same for the metal layer. see my moltenvk PR for example: https://github.com/mpv-player/mpv/pull/7482/files#diff-28b1978e8194b3980f86f4f9213b4c9d0416af71f8bbbe83535ecb6442b306a1R147-R149

same for wantsExtendedDynamicRangeContent  https://developer.apple.com/documentation/quartzcore/cametallayer/1478161-wantsextendeddynamicrangecontent